nft_ingester/src/processors/account_based/mpl_core_fee_indexing_processor.rs

```diff
@@ -26,43 +24,40 @@ pub struct MplCoreFeeProcessor {
     pub metrics: Arc<IngesterMetricsConfig>,
     rpc_client: Arc<RpcClient>,
     rent: Arc<RwLock<Rent>>,
-    join_set: Arc<Mutex<JoinSet<Result<(), tokio::task::JoinError>>>>,
 }
 
 impl MplCoreFeeProcessor {
     pub async fn build(
         storage: Arc<PgClient>,
         metrics: Arc<IngesterMetricsConfig>,
         rpc_client: Arc<RpcClient>,
-        join_set: Arc<Mutex<JoinSet<Result<(), tokio::task::JoinError>>>>,
     ) -> Result<Self, IngesterError> {
         let rent_account = rpc_client.get_account(&rent::ID).await?;
         let rent: Rent = bincode::deserialize(&rent_account.data)?;
-        Ok(Self { storage, metrics, rpc_client, rent: Arc::new(RwLock::new(rent)), join_set })
+        Ok(Self { storage, metrics, rpc_client, rent: Arc::new(RwLock::new(rent)) })
     }
 
     // on-chain programs can fetch rent without RPC call
     // but off-chain indexer need to make such calls in order
     // to get actual rent data
-    pub async fn update_rent(&self, mut rx: Receiver<()>) {
+    pub fn update_rent(&self, cancellation_token: CancellationToken) {
         let rpc_client = self.rpc_client.clone();
         let rent = self.rent.clone();
-        self.join_set.lock().await.spawn(tokio::spawn(async move {
-            while rx.is_empty() {
-                if let Err(e) = Self::fetch_actual_rent(rpc_client.clone(), rent.clone()).await {
-                    error!("fetch_actual_rent: {}", e);
-                    tokio::time::sleep(Duration::from_secs(5)).await;
-                    continue;
-                }
+        usecase::executor::spawn(async move {
+            loop {
```

Comment:
I'm not sure why we need loop and sleep here simultaneously

Reply:
good catch